larpandoracontent/LArTrackShowerId/TrackShowerIdFeatureTool.cc

```diff
@@ -349,6 +360,10 @@ void ThreeDLinearFitFeatureTool::CalculateVariablesSlidingLinearFit(const pandor
                 if (correctedGapLength > maxFitGapLength)
                     maxFitGapLength = correctedGapLength;
             }
+			else
+			{
+				maxFitGapLength = 0.f;
```

Comment:
Perhaps ought to initialise this value to its safe default above, and for *all* arguments you take as non-const reference so that the calling algorithm receives the new value. What you'd then be doing is guarding against all cases where the logic doesn't actually allow you to assign to the variable, which will then just maintain the value it was originally assigned in the calling algorithm.